### PR TITLE
Additional encoder for F7 and added functions for TimerHook for F745 …

### PR DESCRIPTION
**Pull request by robertobucher**
…and H745 and H743

## Summary
We added an additional encoder for F7 and the functions to handle the TimerHook method for sampling time.

The patches add the possibility to use a semphore to handle the sampling time in code automatically generated by pysimCoder, where an additional template makefile is provided for this purpose. This will be possible for the board under stm32f7/nucleo-144, stm32h7/nucleo-h743zi2 and stm32h7/nucleo-h745zi.

In addition, we added a third encoder in the configuration, using TIM8, for the F7 nucleo-144 under stm32f7.

## Impact
boards/arm/stm32f7/nucleo-144/configs/f746-pysim/defconfig 
boards/arm/stm32f7/nucleo-144/include/board.h 
boards/arm/stm32f7/nucleo-144/src/stm32_bringup.c 
boards/arm/stm32h7/nucleo-h743zi2/src/stm32_bringup.c 
boards/arm/stm32h7/nucleo-h745zi/src/stm32_bringup.c

## Testing
Tested on a NUCLEO-F745ZI board

### CODE REVIEW

**Comment by robertobucher** [comment]
I can't understand why the "xtensa-02" failed... Log doesn't help!

**Review by acassis — APPROVED** [review]

**Comment by simbit18** [comment]
Hi @robertobucher  this could just be a temporary network issue
https://github.com/apache/nuttx/actions/runs/10181539722/job/28162043342?pr=12814#step:7:85

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/arm/stm32h7/nucleo-h745zi/src/stm32_bringup.c [hunk 1]
```diff
@@ -227,3 +231,13 @@ int stm32_bringup(void)
 
   return OK;
 }
+
+#ifdef CONFIG_SYSTEMTICK_HOOK
+
+sem_t g_waitsem;
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h745zi/src/stm32_bringup.c:237]: where the semaphore is consumed?

File: boards/arm/stm32h7/nucleo-h745zi/src/nucleo-h745zi.h [hunk 2]
```diff
@@ -317,4 +317,8 @@ int stm32_pwm_setup(void);
 int stm32_qencoder_initialize(const char *devpath, int timer);
 #endif
 
+#ifdef CONFIG_SYSTEMTICK_HOOK
+void board_timerhook(void);
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h745zi/src/nucleo-h745zi.h:321]: prototype is already defined in `board.h`

File: boards/arm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h [hunk 3]
```diff
@@ -226,4 +226,8 @@ int stm32_pwm_setup(void);
 int stm32_qencoder_initialize(const char *devpath, int timer);
 #endif
 
+#ifdef CONFIG_SYSTEMTICK_HOOK
+void board_timerhook(void);
```
Comment (pkarashchenko) [inline boards/arm/stm32h7/nucleo-h743zi2/src/nucleo-h743zi2.h:230]: prototype is already defined in `board.h`

**Review by pkarashchenko — CHANGES_REQUESTED** [review]

File: boards/arm/stm32h7/nucleo-h745zi/src/stm32_bringup.c (hunk 1 above)
Comment (robertobucher) [reply boards/arm/stm32h7/nucleo-h745zi/src/stm32_bringup.c:237]: Semaphore will be consumed in pysimCoder 

File:    nuttx_main_timehook.c

```
[...]
extern sem_t g_waitsem;

[...]
  while(!end){
    while (sem_wait(&g_waitsem) >= 0) {
	if(nTick==0){

[...]
....

**Review by xiaoxiang781216 — CHANGES_REQUESTED** [review]
@robertobucher please drop merge patch from pr and squash the temp patch.

**Review by jerpelea — CHANGES_REQUESTED** [review]
please add a proper title and commit message 

ex
arm/stm32: Additional encoder for F7 and added functions for TimerHook for F7xx

this commit provides.....for F745, H745 and H743

**Pull request closed without merging**